test/e2e/testutils.go

```diff
@@ -0,0 +1,132 @@
+	return &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      catalogSourceConfigName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: targetNamespace,
+			Packages:        packages,
+		}}
+}
+
+// createOperatorSource returns a OperatorSource based on the provided parameters
+func createOperatorSource(operatorSourceName string, namespace string, labels map[string]string, registryNamespace string) *operator.OperatorSource {
+	return &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-operators",
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: registryNamespace,
+		},
+	}
+}
+
+// createRuntimeObject creates a runtime object using the test framework
+func createRuntimeObject(f *test.Framework, ctx *test.TestCtx, obj runtime.Object) error {
+	return f.Client.Create(
+		context.TODO(),
+		obj,
+		&test.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       cleanupTimeout,
+			RetryInterval: cleanupRetryInterval,
+		})
+}
+
+// deleteRuntimeObject deletes a runtime object and wait for up to 1 minute for the
+// object to be removed from the cluster
+func deleteRuntimeObject(f *test.Framework, obj runtime.Object, name string, namespace string) error {
+	// Delete the runtime object
+	err := f.Client.Delete(
+		context.TODO(),
+		obj)
+	if err != nil {
+		return fmt.Errorf("Error deleting obj: %v", err)
+	}
+
+	// Confirm that the runtime object was deleted
+	namespacedName := types.NamespacedName{Name: name, Namespace: namespace}
+	result := pollUntilTrue(5*time.Second, 1*time.Minute, func() bool {
+		err = f.Client.Get(
+			context.TODO(),
+			namespacedName,
+			obj)
+		if apierrors.IsNotFound(err) {
+			return true
+		}
+		return false
+	})
+	if result {
+		return nil
+	}
+	return fmt.Errorf("Runtime object %v was never deleted", obj)
+}
+
+// createNamespace creates a namespace
+func createNamespace(f *test.Framework, ctx *test.TestCtx, namespace string) error {
+	obj := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	// Create the operatorsource to download the manifests
+	return f.Client.Create(
+		context.TODO(),
+		obj,
+		&test.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       cleanupTimeout,
+			RetryInterval: cleanupRetryInterval,
+		})
+}
+
+// pollUntilTrue will check if a function returns a true value
+// every pollInterval duration until the timeout duration has
+// passed
+//
+// pollUntilTrue function will run over the timeout if the
+// provided function hangs
+func pollUntilTrue(pollInterval time.Duration, timeout time.Duration, f func() bool) bool {
```

Comment:
Don't we already have a very similar method to this in helpers.go?

Reply:
In the helpers.go there exists a tool to check that a runtime object exists. There is not a way to check that a runtime Object has reached a certain state.